code/modules/client/preference_setup/loadout/loadout.dm

```diff
@@ -521,7 +553,12 @@
 /decl/loadout_option/proc/spawn_and_validate_item(mob/living/human/H, metadata)
 	PRIVATE_PROC(TRUE)
 
-	var/obj/item/item = spawn_item(H, H, metadata)
+	var/obj/item/item
+	if(apply_to_existing_if_possible)
+		item = locate(path) in H.get_equipped_items(include_carried = TRUE)
```

Comment:
maybe a little out of scope but i feel that there should be support for more complicated predicates, so something like
```dm
		for(var/obj/item/candidate in H.get_equipped_items(include_carried = TRUE))
			if(can_replace_existing(candidate))
				item = candidate
				break
```
and as a default
```dm
/decl/loadout_option/proc/can_replace_existing(obj/item/candidate)
	return istype(candidate, path)
```